File: web-admin/src/services/dashboardService.ts
// Dashboard Service

import { apiRequest } from './api'
import type { DashboardSummary, DashboardFilter } from '@/types/dashboard'

export const dashboardService = {
  // Get dashboard summary with filters
  async getSummary(filters?: DashboardFilter): Promise<DashboardSummary> {
    const params = new URLSearchParams()

    if (filters?.area) params.append('area', filters.area)
    if (filters?.year) params.append('year', filters.year.toString())
    if (filters?.month) params.append('month', filters.month.toString())
    if (filters?.siteCode) params.append('siteCode', filters.siteCode)

    const queryString = params.toString()
    const url = `/dashboard/summary${queryString ? `?${queryString}` : ''}`

    return apiRequest<DashboardSummary>('GET', url)
  },
}

File: PHAInspection/PHAInspection/Services/DashboardService.swift
//
//  DashboardService.swift
//  PHA Inspection App
//
//  Dashboard API Service
//

import Foundation

class DashboardService {
    static let shared = DashboardService()

    private init() {}

    // MARK: - Get Dashboard Summary
    func getDashboardSummary(
        area: String? = nil,
        year: Int? = nil,
        month: Int? = nil,
        siteCode: String? = nil
    ) async throws -> DashboardSummary {
        var endpoint = APIConfig.Endpoints.dashboardSummary + "?"

        var params: [String] = []

        if let area = area {
            params.append("area=\(area)")
        }
        if let year = year {
            params.append("year=\(year)")
        }
        if let month = month {
            params.append("month=\(month)")
        }
        if let siteCode = siteCode {
            params.append("siteCode=\(siteCode)")
        }

        endpoint += params.joined(separator: "&")

        return try await APIService.shared.request(endpoint: endpoint)
    }
}
